lib/outdated.js

```diff
@@ -111,7 +111,12 @@ function outdated (args, silent, cb) {
         }
         output(table(outTable, tableOpts))
       }
-      cb(null, list.map(function (item) { return [item[0].parent.path].concat(item.slice(1, 7)) }))
+      var msg = [
+        'The outdated dependencies were found. Please make sure',
+        'the versions in node_module, or look over package.json',
+        'to use the most suitable semver range.'
+      ].join('\n')
+      cb(msg, list.map(function (item) { return [item[0].parent.path].concat(item.slice(1, 7)) }))
```

Comment:
This isn't an internal error, this is a callback to provide user visible output.

Errors from commands (like this one) are processed via `lib/utils/error-handler.js`, which, if the error argument is a string, prints it to the user using `log.error` and exits with a non-zero exit code.

By contrast, if we called back with an error object (which we often do), then its output message is determined via `lib/utils/error-message.js` and it's logged along with a bunch of other diagnostic information.

This is actually the only way currently to trigger a non-zero exit code without printing all sorts of diagnostic information on folks screens.

If I were to design this, I'd have an error class to use for this kind of scenario and always pass through objects. But IMO that kind of rewrite (while positive) is out of scope for this PR and so I wouldn't ask someone contributing a change like this one to make that kind of rewrite.